src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,88 @@
+import "./Pagination.scss";
+
+interface PaginationProps {
+    currentPage?: number;
+    totalPages?: number;
+    onPageChange: (page: number) => void;
+}
+
+export function Pagination({ currentPage = 1, totalPages, onPageChange }: PaginationProps) {
+    // Проверяем, что totalPages корректно передан (не null и не undefined)
+    if (totalPages == undefined || totalPages <= 0) {
```

Comment:
всегда используйте только строгое сравнение ===
А вообще тут эта проверка излишняя:
```suggestion
    if (!totalPages ) {
```